File: WorkoutApplication.Domain/Exercises/Exercise.cs
using System;
using System.Collections.Generic;
using System.Text;
using WorkoutApplication.Domain.BaseClasses;

namespace WorkoutApplication.Domain.Exercises
{
    public class Exercise : NamedEntity
    {
        public string Description { get; set; }
        public string Setup { get; set; }
        public string Movement { get; set; }
        public string PointsToRemember { get; set; }

        public virtual ICollection<ExerciseTargetAreaLink> ExerciseTargetAreaLinks { get; set; }

        public ResistanceType ResistanceType { get; set; }
        public bool OneSided { get; set; }
    }
}

File: workout-tracker/src/app/workouts/_models/exercise.ts
import { NamedEntity } from '../../shared/models/named-entity';
import { ExerciseTargetAreaLink } from './exercise-target-area-link';

export class Exercise extends NamedEntity {
  publicId: string | null = null; //GUID
  description: string = '';
  exerciseTargetAreaLinks: ExerciseTargetAreaLink[] = [];
  setup: string = '';
  movement: string = '';
  pointsToRemember: string = '';
  resistanceType: number = 0;
  oneSided: boolean = false;
  bandsEndToEnd: boolean | null = null;
  involvesReps: boolean = false;
  usesBilateralResistance: boolean = false;
}
